app/jobs/export_job.rb

```diff
@@ -17,87 +21,71 @@ def perform(user_id, export_type)
-                                          excel_export_monitorees(group),
-                                          "Sara-Alert-Purge-Eligible-Export-Monitorees-#{DateTime.now}-#{index}.xlsx",
-                                          export_type),
-                    filename: "Sara-Alert-Purge-Eligible-Export-Monitorees-#{DateTime.now}-#{index}.xlsx" }
-        lookups << { lookup: save_download(user_id,
-                                          excel_export_assessments(group),
-                                          "Sara-Alert-Purge-Eligible-Export-Assessments-#{DateTime.now}-#{index}.xlsx",
-                                          export_type),
-                    filename: "Sara-Alert-Purge-Eligible-Export-Assessments-#{DateTime.now}-#{index}.xlsx" }
-        lookups << { lookup: save_download(user_id,
-                                          excel_export_lab_results(group),
-                                          "Sara-Alert-Purge-Eligible-Export-Lab-Results-#{DateTime.now}-#{index}.xlsx",
-                                          export_type),
-                    filename: "Sara-Alert-Purge-Eligible-Export-Lab-Results-#{DateTime.now}-#{index}.xlsx" }
-        lookups << { lookup: save_download(user_id,
-                                          excel_export_histories(group),
-                                          "Sara-Alert-Purge-Eligible-Export-Histories-#{DateTime.now}-#{index}.xlsx",
-                                          export_type),
-                    filename: "Sara-Alert-Purge-Eligible-Export-Histories-#{DateTime.now}-#{index}.xlsx" }
+      patients.in_batches(of: RECORD_BATCH_SIZE).each_with_index do |group, index|
+        file_index = index + 1
+        lookups << get_file(user_id, excel_export_monitorees(group), 'Sara-Alert-Purge-Eligible-Export-Monitorees', file_extension, export_type, file_index)
+        lookups << get_file(user_id, excel_export_assessments(group), 'Sara-Alert-Purge-Eligible-Export-Assessments', file_extension, export_type, file_index)
+        lookups << get_file(user_id, excel_export_lab_results(group), 'Sara-Alert-Purge-Eligible-Export-Lab-Results', file_extension, export_type, file_index)
+        lookups << get_file(user_id, excel_export_histories(group), 'Sara-Alert-Purge-Eligible-Export-Histories', file_extension, export_type, file_index)
       end
     end
     return if lookups.empty?
 
+    puts "lookups: #{lookups}"
     # Send an email to user
     UserMailer.download_email(user, export_type, lookups).deliver_later
   end
 
+  # Gets a single download with the provided filename information and containing the provided data.
+  # rubocop:disable Metrics/ParameterLists
+  def get_file(user_id, data, filename, file_extension, export_type, file_index)
+    full_filename = "#{filename}-#{DateTime.now}-#{file_index}.#{file_extension}"
+    { lookup: save_download(user_id, data, full_filename, export_type), filename: full_filename }
+  end
```

Comment:
Could create a second method that just builds the filename and passes that in as 1 parameter to `get_file` to remove extension and index parameters.

Reply:
Agreed, done.